### PR TITLE
(#2175) refactor app.tsx to hooks

### PR DESCRIPTION
**Pull request by samrichca**
**Description of changes**

- refactor app.tsx to hooks
- will be refactoring more in the next ticket - all this does is convert the class component `Root` to a functional component that uses hooks.
- the biggest change is that getInitialProps is now below the component declaration

**Type of Change**

- [ ] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- Design (select one):
  - [ ] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [x] No changes to UI
- UI Kit (select one):
  - [ ] New Components with storybook stories created
  - [ ] Modified Existing components and updates stories
  - [x] No new UIKit components or changes
- [x] Feature is minimally responsive
- [x] Manual testing of UI feature
- [x] Add copyrights to new files
- [x] Connected ticket to PR

### CODE REVIEW

File: pages/_app.tsx [hunk 1]
```diff
@@ -82,235 +82,230 @@ const removeCookie = (res: ServerResponse, cookieName: string) => {
...
+    apolloCache = graphqlQueriesToCache
+      ? await getApolloCacheForQueries(graphqlQueriesToCache)(egoJwt)
+      : {};
+  } catch (e) {
+    console.log(e);
+  }
+
+  const startWithGlobalLoader = Component.startWithGlobalLoader || false;
+
+  const isOauth = checkOauth(ctx);
```
Comment (samrichca) [inline pages/_app.tsx:292]: moved oauth check from constructor to getInitialProps

File: pages/_app.tsx [hunk 2]
```diff
@@ -82,235 +82,230 @@ const removeCookie = (res: ServerResponse, cookieName: string) => {
...
-  constructor(props: any) {
-    super(props);
-    const isOauth = this.isInOauthMode(props);
-    this.state = {
-      isLoadingLoginRedirect: isOauth,
-    };
+Root.getInitialProps = async ({ Component, ctx }: AppContext & { Component: PageWithConfig }) => {
+  const { AUTH_DISABLED, MAINTENANCE_MODE_ON } = getConfig();
+  if (MAINTENANCE_MODE_ON) {
+    return getMaintenanceProps(ctx, MAINTENANCE_MODE_ON);
```
Comment (samrichca) [inline pages/_app.tsx:222]: made this a function for legibility

File: pages/_app.tsx [hunk 3]
```diff
@@ -82,235 +82,230 @@ const removeCookie = (res: ServerResponse, cookieName: string) => {
...
+    ReactGA.initialize(GA_TRACKING_ID, {
+      gaOptions: {
+        userId: userModel ? userModel.context.user.email : 'NA',
+      },
+    });
+    ReactGA.pageview(ctx.asPath);
+  }, []);
+
+  useEffect(() => {
+    async function handleAuth() {
```
Comment (samrichca) [inline pages/_app.tsx:164]: useEffect can't be async, but a function inside of it can

**Review by demariadaniel — APPROVED** [review]
Wow this is a ton of work! Definitely cleans up the file
I've tested a few standard flows on localhost and the app seems to be working normally, well done

**Pull request merged**